internal/plugin/plugin.go

```diff
@@ -93,12 +95,39 @@ func NewPluginClient(connector string, version string, logger hclog.Logger) (Plu
 		return nil, fmt.Errorf("the plugin (%s) doesn't correctly implement the necessary interfaces", connector)
 	}
 
+	currentVersion := version.CurrentCliVersionInformation{}
+	if _, err = isPluginCompatibleWithCli(&currentVersion, is, logger); err != nil {
+		logger.Error("Plugin not compatible: %s", err.Error())
+		return nil, err
+	}
+
 	info := is.PluginInfo()
 	logger.Debug("Using plugin: " + info.String())
 
 	return pci, nil
 }
 
+func isPluginCompatibleWithCli(currentCliInformation version.CliVersionInformation, pluginInfo plugin2.Info, logger hclog.Logger) (bool, error) {
+	if currentCliInformation.GetCliVersion().Equal(version.DevVersion) {
+		logger.Warn("Skip plugin compatibility check for dev version")
+		return true, nil
+	}
+
+	//Check if current CLI version is compatible with CLI version used at plugin compile time (current version > plugin compiled version)
+	pluginCompiledVersion := pluginInfo.CliBuildVersion()
+	if currentCliInformation.CliPluginConstraint().Check(&pluginCompiledVersion) {
+		return true, nil
+	}
+
+	//Check if plugin CLI version is compatible with current CLI version (current version < plugin compiled version)
+	pluginConstraint := pluginInfo.PluginCliConstraint()
```

Comment:
Not sure why this is still needed?
Wouldn't it mean that the plugin always needs to be built with the current CLI version exactly? Maybe I'm not understanding it.